lib/osdep_linux.c

```diff
@@ -101,12 +101,9 @@ void free_mem_obj(mem_obj_t obj)
 
 int copy_pm_table(void *buffer, size_t size)
 {
-	int read_size;
-
 	if(pm_table_fd > 0){
 		lseek(pm_table_fd, 0, SEEK_SET);
-		read_size = read(pm_table_fd, buffer, size);
-		if(read_size == -1) {
+		if((int)read(pm_table_fd, buffer, size) == -1) {
```

Comment:
Hmm, compiler would optimize it so there is no need to eliminate this variable.
It helps with readability.

Reply:
- This depends on the compiler you use (and flags I guess?). On linux by default it is compiled with GCC (std `201710L`) following the commands of the `README.md`. I did some tests before committing and it doesn't seem to get optimized in this case. I will do other tests. (Are there any flags I haven't noticed?
I simply used `-fstack-usage`).
Before: `osdep_linux.c:102:5:copy_pm_table	48	static`
After: `osdep_linux.c:102:5:copy_pm_table	32	static`
Also the access in activation record suggests the same thing:
![Screenshot_20230207_142843](https://user-images.githubusercontent.com/69747171/217257900-32cd5435-a80b-42ee-b7e1-6ae62c3b07a9.png)

- Readability is important. True.


I don't see some flag?
---

Thanks for the reply.

Reply:
`-O2` would help :-D

Reply:
> `-O2` would help :-D

yeah of course. sorry for mistake